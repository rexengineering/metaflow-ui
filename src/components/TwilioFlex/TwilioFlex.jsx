import React from "react";
import PropTypes from "prop-types";
import {makeStyles} from "@material-ui/core";
import clsx from "clsx";

const useStyles = makeStyles(() => ({
    iframe: {
        border: "none",
        width: "100%",
        height: "100%",
    },
    container: {
        width: "100%",
        height: "100vh",
    },
}));

function TwilioFlex({twilioFlexURL, className}){
    const classes = useStyles();
    return (
        <section className={clsx(classes.container, className)}>
            <iframe title="twilioFlex" className={classes.iframe} src={twilioFlexURL} />
        </section>
    );
}

TwilioFlex.defaultProps = {
  className: "",
}

TwilioFlex.propTypes = {
  twilioFlexURL: PropTypes.string.isRequired,
  className: PropTypes.string,
}

export default TwilioFlex;