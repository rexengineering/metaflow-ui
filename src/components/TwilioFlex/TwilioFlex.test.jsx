import { render } from "@testing-library/react";
import React from "react";
import TwilioFlex from "./TwilioFlex";

describe("<TwilioFlex />", () => {
    const renderTwilioFlex = (url = "https://flex.twilio.com/agent-desktop/") => (
        <TwilioFlex twilioFlexURL={url} />
    );

    it("should render", () => {
        const { asFragment } = render(renderTwilioFlex());
        const initialRender = asFragment();

        expect(initialRender).not.toBeNull();
    });

    it("should match snapshot", () => {
        const { asFragment } = render(renderTwilioFlex());
        const initialRender = asFragment();

        expect(initialRender).toMatchSnapshot();
    });

    it("should render an iframe", () => {
        render(renderTwilioFlex());
        const iframe = document.querySelector('iframe');
        expect(iframe).toBeInTheDocument();
    });
});
