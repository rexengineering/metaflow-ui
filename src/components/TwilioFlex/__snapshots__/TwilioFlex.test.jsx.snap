// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TwilioFlex /> should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="makeStyles-container-4"
  >
    <iframe
      class="makeStyles-iframe-3"
      src="https://flex.twilio.com/agent-desktop/"
      title="twilioFlex"
    />
  </section>
</DocumentFragment>
`;
